hydrolib/core/dflowfm/extold/models.py

```diff
@@ -0,0 +1,417 @@
+    SpiderWebData = 5
+    """5. Spiderweb data (cyclones)"""
+    CurvilinearData = 6
+    """6. Curvilinear data"""
+    Samples = 7
+    """7. Samples"""
+    TriangulationMagnitudeAndDirection = 8
+    """8. Triangulation magnitude and direction"""
+    Polyline = 9
+    """9. Polyline (<*.pli>-file)"""
+    NetCDFGridData = 11
+    """11. NetCDF grid data (e.g. meteo fields)"""
+    NetCDFWaveData = 14
+    """14. NetCDF wave data"""
+
+
+class Method(IntEnum):
+    """Enum class containing the valid values for the `method` attribute
+    in the [ExtForcing][hydrolib.core.dflowfm.extold.models.ExtForcing] class.
+    """
+
+    PassThrough = 1
+    """1. Pass through (no interpolation)"""
+    InterpolateTimeAndSpace = 2
+    """2. Interpolate time and space"""
+    InterpolateTimeAndSpaceSaveWeights = 3
+    """3. Interpolate time and space, save weights"""
+    InterpolateSpace = 4
+    """4. Interpolate space"""
+    InterpolateTime = 5
+    """5. Interpolate time"""
+    Averaging = 6
+    """6. Averaging"""
+    InterpolateExtrapolateTime = 7
+    """7. Interpolate/Extrapolate time"""
+
+
+class Operand(str, Enum):
+    """Enum class containing the valid values for the `operand` attribute
+    in the [ExtForcing][hydrolib.core.dflowfm.extold.models.ExtForcing] class.
+    """
+
+    OverwriteExistingValues = "O"
+    """Existing values are overwritten."""
+    SuperimposeNewValues = "+"
+    """New values are superimposed."""
+
+
+class ExtForcing(BaseModel):
+    """Class holding the external forcing values."""
+
+    quantity: Union[Quantity, str] = Field(alias="QUANTITY")
+    """Union[Quantity, str]: The name of the quantity."""
+
+    filename: Union[PolyFile, DiskOnlyFileModel] = Field(None, alias="FILENAME")
+    """Union[PolyFile, DiskOnlyFileModel]: The file associated to this forcing."""
+
+    varname: Optional[str] = Field(None, alias="VARNAME")
+    """Optional[str]: The variable name used in `filename` associated with this forcing; some input files may contain multiple variables."""
+
+    sourcemask: DiskOnlyFileModel = Field(
+        default_factory=lambda: DiskOnlyFileModel(None), alias="SOURCEMASK"
+    )
+    """DiskOnlyFileModel: The file containing a mask."""
+
+    filetype: FileType = Field(alias="FILETYPE")
+    """FileType: Indication of the file type.
+    
+    Options:
+    1. Time series
+    2. Time series magnitude and direction
+    3. Spatially varying weather
+    4. ArcInfo
+    5. Spiderweb data (cyclones)
+    6. Curvilinear data
+    7. Samples (C.3)
+    8. Triangulation magnitude and direction
+    9. Polyline (<*.pli>-file, C.2)
+    11. NetCDF grid data (e.g. meteo fields)
+    14. NetCDF wave data
+    """
+
+    method: Method = Field(alias="METHOD")
+    """Method: The method of interpolation.
+    
+    Options:
+    1. Pass through (no interpolation)
+    2. Interpolate time and space
+    3. Interpolate time and space, save weights
+    4. Interpolate space
+    5. Interpolate time
```

Comment:
Should `Averaging` also be in the documentation here?

Reply:
Yes!